include/AutomatableModel.h

```diff
@@ -435,6 +437,13 @@ template <typename T> class LMMS_EXPORT TypedAutomatableModel : public Automatab
 		return AutomatableModel::value<T>( frameOffset );
 	}
 
+	float valueAt(size_t index, int frameOffset = 0)
+    {
+		const auto buffer = valueBuffer();
+		size_t clampedIndex = std::clamp<size_t>(index, 0, buffer->size());
+		return buffer ? (*buffer)[clampedIndex] : value(frameOffset);
```

Comment:
@messmerd @sakertooth is this clamping fine?

Reply:
What if the buffer is null, wouldn't the `buffer->size()` call crash or worse? 
OTOH, if it's guaranteed never to be null, I think `return (*buffer)[clampedIndex]` should be enough.

Reply:
I'm not sure the buffer won't be null. This function was introduced to reduce the redundancy in many places regarding the computation of values.

Reply:
This was a @sakertooth idea

Reply:
Why do we need to clamp here?

Reply:
To avoid writing beyond the bounds

Reply:
I could remove it but don't wanna create any regressions.

Reply:
I think you should try building it locally without the clamping, then test it. And if it seems fine, keep it removed but just add an `assert` to make debug builds bounds checked.

There's a good chance some of the clamps used in this PR are unnecessary. If so, they would potentially be covering up bugs that we should instead be catching with `assert`.

Reply:
I have changed up the order to use multiple returns.